### DISCUSSION TITLE
Timed_out error - how to get file_id

### DISCUSSION THREAD

**Discussion by Javokhir-1999** [python-telegram-bot/python-telegram-bot#2951] [Q&A] [answered]
Hi, I have the List of `Universities ` with their `Images ` in my DB, I am getting and sending them in this way:

    univer = getDataByTitleVal(Universities, 'INHA')
	images = getUninersityImages(univer)
	list = images
	botGlob.sendMediaGroup(
		chat_id=update.effective_chat.id,
		media=[InputMediaPhoto(open('media/'+list[0][0], 'rb'),caption=univer.desc)] +
			[InputMediaPhoto(open('media/'+list[i][0], 'rb')) for i in range(1, len(list))
		]
        )
All works, but Some times it works really slow and even gives me a 'Timed Out' Error. 
guess That is because Images some times heavy or the process itself too complicated,
Can someone advise me on how can I make it better or at least avoid Errors?

And One more thing, I have an idea to save Image on the telegram server before and then use  `file_id ` instead of InputFile every time, but how can I get my 'file_id' after upload?.

Hope for any help

**Comment by harshil21** [comment, marked as answer]
To get the `file_id` of an image, see [this section](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Code-snippets#posting-files), which shows you how to get a file id. 

For timeout errors, you can try increasing the timeouts as specified in [this section](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Handling-network-errors#urllib3-tweaks), note that you can also increase the timeout for [send_media_group](https://python-telegram-bot.readthedocs.io/en/master/telegram.bot.html#telegram.Bot.send_media_group)

**Reply by Javokhir-1999** [reply]
Oh, thanks a lot, I will try them all
